processor/src/operations/decorators.rs

```diff
@@ -7,9 +11,79 @@ impl Process {
     // DEBUGGING
     // --------------------------------------------------------------------------------------------
 
-    /// TODO: implement
-    pub fn op_debug(&self) -> Result<(), ExecutionError> {
-        unimplemented!();
+    /// Print out debugging information based on options passed
+    pub fn op_debug(&mut self, options: DebugOptions) -> Result<(), ExecutionError> {
+        match options {
+            DebugOptions::All => {
+                self.print_stack(None);
+                self.print_mem(None, None);
+            }
+            DebugOptions::Stack(n) => self.print_stack(n),
+            DebugOptions::Mem(n, m) => self.print_mem(n, m),
+            DebugOptions::Local() => self.print_local(),
+        }
+
+        Ok(())
+    }
+
+    // DEBUG HELPER
+    // ---------------------------------------------------------------------------------
+
+    /// Print stack information for debugging.
+    /// If n is passed, this prints the last n items in the stack.
+    fn print_stack(&self, n: Option<usize>) {
+        let states = self.stack.get_states(n);
+        for (i, value) in states.iter().enumerate() {
+            info!("stack [ index: {:?}, value: {:?} ]", i, value.as_int());
+        }
+    }
+
+    /// Create the display string for printing a word.
+    /// Prints <empty> if no value exists.
+    fn fmt_word(&self, word: Option<Word>) -> String {
```

Comment:
does this need to be a function on `self`? It seems like we don't use `self` here.